syntax = "proto3";

package aas.ipc;

// The Bridge service definition.
service Bridge {
  // Sends a high-level command from AAS to Maelstrom
  rpc ExecuteCommand (CommandRequest) returns (CommandResponse) {}
  
  // Streams game state snapshots from Maelstrom to AAS
  rpc StreamSnapshots (SnapshotRequest) returns (stream SnapshotResponse) {}

  // Subscribe to real-time task updates
  rpc SubscribeToTasks (TaskSubscriptionRequest) returns (stream TaskUpdate) {}
}

message TaskSubscriptionRequest {
  string client_id = 1;
}

message TaskUpdate {
  string task_id = 1;
  string title = 2;
  string status = 3;
  string assignee = 4;
  string event_type = 5; // e.g., "CREATED", "CLAIMED", "COMPLETED", "UPDATED"
  int64 timestamp = 6;
}

message CommandRequest {
  string command_type = 1;
  string payload_json = 2;
}

message CommandResponse {
  bool success = 1;
  string message = 2;
}

message SnapshotRequest {
  int32 interval_ms = 1;
}

message SnapshotResponse {
  string snapshot_json = 1;
  int64 timestamp = 2;
}
